### DISCUSSION TITLE
Kyuubi 1.1.0 release note

### DISCUSSION THREAD

**Discussion by ulysses-you** [apache/kyuubi#415] [General]
# Changelog

## v1.1.0-rc3 (09/03/2021)
*No changelog for this release.*

---

#### Bug

- [**Bug**][**Test**] Error running 'select function' command [#397](https://github.com/yaooqinn/kyuubi/issues/397)

---

#### Bug

- [**Bug**] Can not access hivevar set by `beeline --hivevar`  [#383](https://github.com/yaooqinn/kyuubi/issues/383)
- [**Bug**] FetchOrientation.FETCH_FIRST is not handled properly [#360](https://github.com/yaooqinn/kyuubi/issues/360)

#### Feature

- [**Feature**] [Feature Request] Support run kyuubi-server in the foreground [#312](https://github.com/yaooqinn/kyuubi/issues/312)
- [**Feature**] GetTables supports DSv2 and keeps its backward compatibility [#309](https://github.com/yaooqinn/kyuubi/issues/309)

#### closed

- [**closed**] Move to hadoop shaded client [#398](https://github.com/yaooqinn/kyuubi/issues/398)
- [**closed**] Clean up discovery service before engine stop in connection share level [#386](https://github.com/yaooqinn/kyuubi/issues/386)
- [**closed**] Release session if shared level is CONNECTION [#381](https://github.com/yaooqinn/kyuubi/issues/381)
- [**closed**] Auto upload test log in GitHub Actions [#372](https://github.com/yaooqinn/kyuubi/issues/372)
- [**closed**] THRIFT-4858 causes metadata  operations being flaky [#368](https://github.com/yaooqinn/kyuubi/issues/368)
- [**closed**] Log Framework conflict [#363](https://github.com/yaooqinn/kyuubi/issues/363)
- [**closed**] Avoid using HiveResult [#350](https://github.com/yaooqinn/kyuubi/issues/350)
- [**closed**] Add Spark 3.1 profile [#298](https://github.com/yaooqinn/kyuubi/issues/298)

#### dependencies

- [**dependencies**] Make distribution bundle Spark optional [#377](https://github.com/yaooqinn/kyuubi/issues/377)

#### enhancement

- [**enhancement**][**experimental**] GetColumns supports DSv2 and keeps its backward compatibility [#310](https://github.com/yaooqinn/kyuubi/issues/310)

#### question

- [**question**] DBeaver try conect 'jdbc:hive2//ip:port' fail [#384](https://github.com/yaooqinn/kyuubi/issues/384)
- [**question**] Does session have a timeout cleanup mechanism? [#347](https://github.com/yaooqinn/kyuubi/issues/347)

**Comment by ulysses-you** [comment]
cc @pan3793 @yaooqinn

**Comment by ulysses-you** [comment]
This note created by https://github.com/github-tools/github-release-notes

**Comment by pan3793** [comment]
It almost looks good! Only one thing I noticed that `Bug` `Feature` start with UPPER CASE, but others start with lower case. Can we align them?

**Reply by ulysses-you** [reply]
Seems it comes from the github label, which has different case. We may need cleanup it manul if we want to do this ..

**Comment by yaooqinn** [comment]
https://github.com/yaooqinn/kyuubi/milestone/13?closed=1 why the list is smaller than 1.1 milestone

**Reply by ulysses-you** [reply]
I guess it only collect the issue, without pr.

**Comment by yaooqinn** [comment]
It looks ugly :(

how about `glog`



* 471dbbc (HEAD -> master) [KYUUBI #411] [DOC] Access Kyuubi with Hive JDBC and ODBC Drivers
* 3eefea6 [KYUUBI #410] Add Spark-3.1 trouble shooting
* 57ed76f [KYUUBI #408] Fix engine log does not be overwrite
* e971732 [KYUUBI #407] Auto update release note
* e2fd967 [KYUUBI #397] Add test case for querying columns with dots
* 016277d [KYUUBI #406] Create GitHub Release only once
* 5e2a129 Prepare Kyuubi v1.2.0
* 183e18f [KYUUBI #404] How To Use Spark Dynamic Resource Allocation (DRA) in Kyuubi
* 03b7112 (tag: v1.1.0-rc1, tag: v1.1.0-rc0, tag: v1.1) [KYUUBI #403] FIx flaky test in ConnectionLevelSparkEngineSuite
* 9cb507d [KYUUBI #395] [WIP] FIx flaky test in ConnectionLevelSparkEngineSuite
* 352b4eb [KYUUBI #400] Implement a simpler toHiveString to convert spark row
* 8e83245 [KYUUBI #402] Remove hadoop suffix from tarball name when dist —without-spark
* 562c458 [KYUUBI #398][FOLLOWUP] exclude unused jars
* 4abc3ff [KYUUBI #398] Move to hadoop shaded client
* 8dd54a2 [KYUUBI #383] Support beeline --hivevar --hiveconf
* 874c71c [KYUUBI #396] Add Backward Compatibility Checker for Compile with Spark 3.0 but Run on 3.1
* 25d2d58 [KYUUBI #393] Add Support for Hadoop 3.2
* 2e33158 [KYUUBI #298]Add Support for Apache Spark 3.1.1
* aa878e4 [KYUUBI #386] Clean up discovery service before engine stop in connection share level
* 8e8679e [KYUUBI #392] Remove netty dependency
* d10b17d [KYUUBI #391] Bump up guava to 14.0.1
* e49ff88 [KYUUBI #390] Replace abbr in doc
* f83faa8 [KYUUBI #389] Bump up curator to 2.12.0
* 3d0f050 [KYUUBI #388] How To Use Spark Adaptive Query Execution (AQE) in Kyuubi

**Comment by ulysses-you** [comment]
I just find some config, how about this 



# Changelog

## v1.1.0 (10/03/2021)
- [[KYUUBI #413] [MINOR]: remove unused class import](https://github.com/yaooqinn/kyuubi/commit/8f44d0b4cf84593892fb1f8d59ae0fd9f9493c5e) - @MyLanPangzi
- [[KYUUBI #412] [MINOR] Simplify ProcBuilder code](https://github.com/yaooqinn/kyuubi/commit/4d3f0742dd0bf4f0b6c09214d8a79744c9afea56) - @MyLanPangzi
- [[KYUUBI #411] [DOC] Access Kyuubi with Hive JDBC and ODBC Drivers](https://github.com/yaooqinn/kyuubi/commit/471dbbcb6185f2270914b3b02dabe92bfac0fe97) - @yaooqinn

---

## v1.1.0-rc3 (09/03/2021)
- [[KYUUBI #408] Fix engine log does not be overwrite](https://github.com/yaooqinn/kyuubi/commit/57ed76f48d7f30fe7df317920fde57d95e92326e) - @ulysses-you

---

## v1.1.0-rc2 (08/03/2021)
- [[KYUUBI #397] Add test case for querying columns with dots](https://github.com/yaooqinn/kyuubi/commit/e2fd967c58e885fe7401baebe25e8943d5b970d4) - @Honglun
- [[KYUUBI #406] Create GitHub Release only once](https://github.com/yaooqinn/kyuubi/commit/016277d183e8d0f8727273c29dcbcfb113b56d29) - @yaooqinn
- [Prepare Kyuubi v1.2.0](https://github.com/yaooqinn/kyuubi/commit/5e2a1290c3e26be426d73ff51aa5c3ee4c56a232) - @yaooqinn

---

## v1.1.0-rc1 (08/03/2021)
- [[KYUUBI #403] FIx flaky test in ConnectionLevelSparkEngineSuite](https://github.com/yaooqinn/kyuubi/commit/03b7112eee5a87da059e22716ea54598e5b76bed) - @ulysses-you
- [[KYUUBI #395] [WIP] FIx flaky test in ConnectionLevelSparkEngineSuite](https://github.com/yaooqinn/kyuubi/commit/9cb507d3d1ecf903c1bbe3bb5fc04a25ba2ece7b) - @ulysses-you
- [[KYUUBI #400] Implement a simpler toHiveString to convert spark row](https://github.com/yaooqinn/kyuubi/commit/352b4eb99e95a3a9eb091ae591d6885f72d5dc4f) - @yaooqinn
- [[KYUUBI #402] Remove hadoop suffix from tarball name when dist —without-spark](https://github.com/yaooqinn/kyuubi/commit/8e83245d7a13ddd85033ba9cb33f17828db99086) - @pan3793
- [[KYUUBI #398][FOLLOWUP] exclude unused jars](https://github.com/yaooqinn/kyuubi/commit/562c45865fb1cb16b79c68843fafd2c7523f9f1b) - @pan3793
- [[KYUUBI #398] Move to hadoop shaded client](https://github.com/yaooqinn/kyuubi/commit/4abc3ff6a02757759eeb10a78549502ed80163ba) - @pan3793
- [[KYUUBI #383] Support beeline --hivevar --hiveconf](https://github.com/yaooqinn/kyuubi/commit/8dd54a29b3aa593a056f10a6b585d6b9dd41f879) - @pan3793
- [[KYUUBI #396] Add Backward Compatibility Checker for Compile with Spark 3.0 but Run on 3.1](https://github.com/yaooqinn/kyuubi/commit/874c71c6fe1160e51158ce2c970863961e5d832d) - @yaooqinn
- [[KYUUBI #393] Add Support for Hadoop 3.2](https://github.com/yaooqinn/kyuubi/commit/25d2d587b018a6456b6ea78171ce4f84025d7a7a) - @yaooqinn
- [[KYUUBI #298]Add Support for Apache Spark 3.1.1](https://github.com/yaooqinn/kyuubi/commit/2e3315867beedac6a5c685dd0620f1edd1d83431) - @yaooqinn
- [[KYUUBI #386] Clean up discovery service before engine stop in connection share level](https://github.com/yaooqinn/kyuubi/commit/aa878e4d4826cbba92d29f466a778fd04d0e2fe5) - @ulysses-you
- [[KYUUBI #392] Remove netty dependency](https://github.com/yaooqinn/kyuubi/commit/8e8679e92bf7a0e175895bfb6177330a8c2b176f) - @yaooqinn
- [[KYUUBI #391] Bump up guava to 14.0.1](https://github.com/yaooqinn/kyuubi/commit/d10b17d87f67179f9c9630142b52e3c08855fb5f) - @yaooqinn
- [[KYUUBI #390] Replace abbr in doc](https://github.com/yaooqinn/kyuubi/commit/e49ff8830567ae77d4d7c2360b244bf068eef63a) - @pan3793
- [[KYUUBI #389] Bump up curator to 2.12.0](https://github.com/yaooqinn/kyuubi/commit/f83faa8356dca2717ab8970d887319d9c677bd5e) - @yaooqinn
- [[KYUUBI #388] How To Use Spark Adaptive Query Execution (AQE) in Kyuubi](https://github.com/yaooqinn/kyuubi/commit/3d0f050034b7d8b8bf4f370905e014fef556f798) - @yaooqinn
- [[KYUUBI #385] Improve pass config between kyuubi and spark engine](https://github.com/yaooqinn/kyuubi/commit/4eed2ee0a718b273b12eba36610e1563a0504c3d) - @ulysses-you
- [[KYUUBI #381] Release session if shared level is CONNECTION](https://github.com/yaooqinn/kyuubi/commit/7f2a24bd9f25d05df6f3a587bc466994f72176c7) - @ulysses-you
- [[KYUUBI #380] [FOLLOWUP #377] change default distribution bundle name, and distribute without spark bundle](https://github.com/yaooqinn/kyuubi/commit/08dbe24d28a6f28d8073e604c8258e3966923b9b) - @pan3793
- [[KYUUBI #379] Improve engine log capture](https://github.com/yaooqinn/kyuubi/commit/0f6256817107fbb127e7556dff4bb612f07eb0d0) - @yaooqinn
- [[KYUUBI #377] build/dist support --spark-provided](https://github.com/yaooqinn/kyuubi/commit/2f8df2355dcdc73d8e4a87447f6ecc5972e43426) - @pan3793
- [[KYUUBI #376] Use spark-hive instead of spark-hive-thriftserver in engine test](https://github.com/yaooqinn/kyuubi/commit/968ed4f07919a893633e4b19ab6307f2035a3ab1) - @yaooqinn
- [[KYUUBI #375] THRIFT-4858 causes metadata operations being flaky](https://github.com/yaooqinn/kyuubi/commit/0a315f3287ea7467647fe7c2784654f815c9b89f) - @yaooqinn
- [[KYUUBI #374] [DOC] Add Collaborators page](https://github.com/yaooqinn/kyuubi/commit/da8d2d4dd3b1d3d3cfb90dfb6ffcb79e784ed6de) - @pan3793
- [[KYUUBI #372] Auto upload unit-tests.log](https://github.com/yaooqinn/kyuubi/commit/81e36a13894f1489129a28c53a513ea3a3859e8d) - @pan3793
- [[KYUUBI #371] Avoid creating useless operation log for meta operation](https://github.com/yaooqinn/kyuubi/commit/ce7b122435c143d2cf3c6aaba84dd32c4d83bc93) - @yaooqinn
- [[KYUUBI #360] Correct handle getNextRowSet with FETCH_PRIOR FETCH_FIRST](https://github.com/yaooqinn/kyuubi/commit/c659089bc22a032d41dcd4e741adf56a5a82ea63) - @pan3793
- [[KYUUBI #369] Using a lightweight zk client for kyuubi session](https://github.com/yaooqinn/kyuubi/commit/d94b1c4519662b12ca849e15551add08b37d99a8) - @yaooqinn
- [[KYUUBI #367] Engine terminating checker should start after exposed](https://github.com/yaooqinn/kyuubi/commit/4b13f1c551507e28307077594f33f577c6c228f8) - @yaooqinn
- [[KYUUBI #365] Support user specify app name and tag](https://github.com/yaooqinn/kyuubi/commit/2ff851f9bf7204952f4fd1afd79cd351bd388d27) - @ulysses-you
- [[KYUUBI #363] exclude hive-shims unused dependencies](https://github.com/yaooqinn/kyuubi/commit/399b30bb952c3627d573b471e5b1dbc52418ea1a) - @pan3793
- [[KYUUBI #361] Remove Hive Suffix for built in](https://github.com/yaooqinn/kyuubi/commit/28db93cc7596d1785afd3398b9828dd369a5048d) - @yaooqinn
- [[KYUUBI #310] GetColumns supports DSv2 and keeps its backward compatibility](https://github.com/yaooqinn/kyuubi/commit/ee50890b00b1982cca331e224e1bb3e96db0f044) - @yaooqinn
- [[KYUUBI #356] Fix missing default versions](https://github.com/yaooqinn/kyuubi/commit/54c306b9671a7d18deb3ff7be0d710c8f863a793) - @yaooqinn
- [[KYUUBI #356] Update built-in Spark version to 3.0.2](https://github.com/yaooqinn/kyuubi/commit/bacff6c59a31bf72ab4a83f810587b881abfbfb8) - @ulysses-you
- [[KYUUBI #347] The 'lastIdleTime' needs to be updated after the last expired Handle has been cleaned](https://github.com/yaooqinn/kyuubi/commit/681b0d188dbfb85fa7bfc2d5a7782d88aa59cfa7) - @Honglun
- [[KYUUBI #350] Avoid using HiveResult](https://github.com/yaooqinn/kyuubi/commit/23e1d61e47a4f45f95e4ea6a5bc0d271c7b6603d) - @pan3793
- [[KYUUBI #312] Support run kyuubi in foreground](https://github.com/yaooqinn/kyuubi/commit/600b014070d3f1f4d11881473a7af4b4f78aa3f1) - @pan3793
- [[KYUUBI #349] Fix typo](https://github.com/yaooqinn/kyuubi/commit/5637e9105e31dc3b4a2d7f58106e76db3bebbd89) - @yangrong688
- [[KYUUBI #309] GetTables supports DSv2 and keeps its backward compatibility](https://github.com/yaooqinn/kyuubi/commit/a2818003e68e3bcd7f8c0038d44100c82a82a67c) - @yaooqinn
- [[KYUUBI #347]Closing expired operations does not require updating 'lastAccessTime'](https://github.com/yaooqinn/kyuubi/commit/c1a71b7d00ddce0b278386200cb2e0ba8a459d62) - @Honglun

**Reply by yaooqinn** [reply]
LGTM

**Reply by yaooqinn** [reply]
Is it able to group them to `Feature/Bug/Doc/Others`

**Reply by yaooqinn** [reply]
some patch seems belong to the  master branch

**Reply by ulysses-you** [reply]
> Is it able to group them to Feature/Bug/Doc/Others

A liitle hard. This log is based on commits not issues.

> some patch seems belong to the master branch

Seems it get commits use the tag date ..

**Comment by yaooqinn** [comment]
I'd like our release note template to be like this (we can keep update!)

# Anouncing Kyuubi 1.1.0

Part.1  - Greeting
 Hi, xxxxx...

 - release tag link
 - Doc link
 - milestone link

## HighLighting

Part.2 - HighLighting, manually highlight some PR that brings remarkable feature, such as Spark 3.1 support

## Breaking Changes
Part.3 - Breaking Changes, manually add some PR that breaks with the last stable release if any

## Features

Part 4 autogen

## Bugs

Part 5 autogen


Thanks

WDYT?

**Reply by ulysses-you** [reply]
Yes, we can do this.  It's so kind to provide the clear note and it's always useful.

Otherhand, I believe almost people won't read the commit log.

**Reply by yaooqinn** [reply]
any update on this？

**Reply by ulysses-you** [reply]
Has updated a rough note..

**Comment by yaooqinn** [comment]
I've updated the release note, please review thanks

**Reply by pan3793** [reply]
LGTM, and I tuned some spaces in HighLighting

**Reply by Honglun** [reply]
There’s no space between some issue id and the title, e.g. [KYUUBI #307]GetCatalogs, [KYUUBI #347]Closing, [KYUUBI #298]Add

**Reply by pan3793** [reply]
Let me recheck

**Reply by pan3793** [reply]
Reformatted, and tuned some descriptions, it should look better now.
